src/components/UserConsent/DeclineModal.tsx

```diff
@@ -77,7 +78,7 @@ const DeclineModal: FunctionComponent<DeclineModalProps> = ({
       role={'dialog'}
       shouldCloseOnOverlayClick={true}
       closeTimeoutMS={MODAL_ANIMATION_DURATION}
-      appElement={containerEl}
+      appElement={containerEl || undefined}
```

Comment:
This is something I sometimes struggle with, but how should we decide when/where to use `null` vs `undefined`? 

Like in this case at line 18 there is `containerEl: HTMLElement | null` and here it's `appElement={containerEl || undefined}`

Reply:
Yeah `null` and `undefined` are often confused with each other. My rules of thumb:

- Prefer `undefined` to `null`, because it presents the absent state, while `null` means there's an object there but it's value is `null`.
- `undefined` is better for object destructuring, e.g.:
```javascript
const obj = { foo: null }
const { foo = 1 } = obj  // foo is `null`
const { bar = 2 } = obj  // bar is `2`
```
- `null` is the [billion dollar mistake](https://hackernoon.com/null-the-billion-dollar-mistake-8t5z32d6), so just keep away from it